### PR TITLE
[release-4.12] OCPBUGS-6841: Update the vertical scaling test to account for CPMSO

### PR DESCRIPTION
**Pull request by Elbehery**
This PR is a manual cherrypick of https://github.com/openshift/origin/pull/27788

resolves https://issues.redhat.com/browse/OCPBUGS-6841

### CODE REVIEW

**Review by geliu2016 — APPROVED** [review]
/label cherry-pick-approved

**Comment by Elbehery** [comment]
locally `make verify` 

```
hack/verify-jsonformat.sh
hack/verify-generated.sh
SUCCESS after 47.000s: hack/verify-generated.sh:13: executing '/Users/mustafa/workspace/go/src/openshift/origin/hack/update-generated.sh' expecting success
SUCCESS after 0.000s: hack/verify-generated.sh:14: executing 'git diff --exit-code /Users/mustafa/workspace/go/src/openshift/origin/test/extended/util/annotate/generated/' expecting success
SUCCESS after 1.000s: hack/verify-generated.sh:15: executing 'git diff --exit-code /Users/mustafa/workspace/go/src/openshift/origin/test/extended/util/image/zz_generated.txt' expecting success
[INFO] hack/verify-generated.sh exited with code 0 after 00h 00m 48s
```

dont know why it fails here 

```
ERRO[2023-05-05T08:57:15Z] 
  * could not run steps: step verify failed: test "verify" failed: could not watch pod: the pod ci-op-75zgjrnz/verify failed after 4m25s (failed containers: test): ContainerFailed one or more containers exited
Container test exited with code 2, reason Error
---
-51457,11 +51457,13 @@ var _bindata = map[string]func() (*asset
 // directory embedded in the file by go-bindata.
 // For example if you run go-bindata on data/... and data contains the
 // following hierarchy:
-//     data/
-//       foo.txt
-//       img/
-//         a.png
-//         b.png
+//
+//	data/
+//	  foo.txt
+//	  img/
+//	    a.png
+//	    b.png
+//
 // then AssetDir("data") would return []string{"foo.txt", "img"}
 // AssetDir("data/img") would return []string{"a.png", "b.png"}
 // AssetDir("foo.txt") and AssetDir("notexist") would return an error
make: *** [Makefile:101: verify-bindata-bindata] Error 1
hack/verify-jsonformat.sh
hack/verify-generated.sh
Running hack/verify-generated.sh:13: executing '/go/src/github.com/openshift/origin/hack/update-generated.sh' expecting success...
SUCCESS after 37.[67](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27907/pull-ci-openshift-origin-release-4.12-verify/1654403481483612160#1:build-log.txt%3A67)0s: hack/verify-generated.sh:13: executing '/go/src/github.com/openshift/origin/hack/update-generated.sh' expecting success
Running hack/verify-generated.sh:14: executing 'git diff --exit-code /go/src/github.com/openshift/origin/test/extended/util/annotate/generated/' expecting success...
SUCCESS after 1.4[76](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27907/pull-ci-openshift-origin-release-4.12-verify/1654403481483612160#1:build-log.txt%3A76)s: hack/verify-generated.sh:14: executing 'git diff --exit-code /go/src/github.com/openshift/origin/test/extended/util/annotate/generated/' expecting success
Running hack/verify-generated.sh:15: executing 'git diff --exit-code /go/src/github.com/openshift/origin/test/extended/util/image/zz_generated.txt' expecting success...
SUCCESS after 0.034s: hack/verify-generated.sh:15: executing 'git diff --exit-code /go/src/github.com/openshift/origin/test/extended/util/image/zz_generated.txt' expecting success
[INFO] hack/verify-generated.sh exited with code 0 after 00h 00m 39s
make: Target 'verify' not remade because of errors.
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"k8s.io/test-infra/prow/entrypoint/run.go:[79](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27907/pull-ci-openshift-origin-release-4.12-verify/1654403481483612160#1:build-log.txt%3A79)","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2023-05-05T08:57:14Z"}
--- 
INFO[2023-05-05T08:57:15Z] Reporting job state 'failed' with reason 'executing_graph:step_failed:running_pod'

```

File: test/extended/testdata/bindata.go [hunk 1]
```diff
@@ -51457,13 +51457,11 @@ var _bindata = map[string]func() (*asset, error){
...
 // For example if you run go-bindata on data/... and data contains the
 // following hierarchy:
-//
-//	data/
-//	  foo.txt
-//	  img/
-//	    a.png
-//	    b.png
-//
+//     data/
```
Comment (Elbehery) [inline test/extended/testdata/bindata.go:51460]: @tjungblu @hasbro17 @DennisPeriquet Hello ✋🏽 

the `make-verify` ci job fails and I tried to remove this or add it, it is always the same !!!

any thoughts ?

**Comment by Elbehery** [comment]
running make verify locally 

```
 make verify
Running `gofmt -s -l` on 518 file(s).
go vet -mod=vendor ./...
ln -s /Users/mustafa/workspace/go/src/openshift/origin/vendor "/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.PUz3EMOo/src"
export GO111MODULE=off && export GOPATH=/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.PUz3EMOo && export GOBIN=/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.PUz3EMOo/bin && go install "./vendor/github.com/go-bindata/go-bindata/..."
/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.PUz3EMOo/bin/go-bindata -nocompress -nometadata -prefix "testextended" -pkg "testdata" -o "/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.2ZyrVERe/test/extended/testdata/bindata.go" -ignore "OWNERS" -ignore ".*\.(go|md)$" examples/db-templates examples/image-streams examples/sample-app examples/quickstarts/... examples/hello-openshift examples/jenkins/... examples/quickstarts/cakephp-mysql.json test/extended/testdata/... e2echart && gofmt -s -w "/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.2ZyrVERe/test/extended/testdata/bindata.go" && diff -Naup {.,/var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/tmp.2ZyrVERe}/test/extended/testdata/bindata.go
hack/verify-jsonformat.sh
hack/verify-generated.sh
SUCCESS after 47.000s: hack/verify-generated.sh:13: executing '/Users/mustafa/workspace/go/src/openshift/origin/hack/update-generated.sh' expecting success
SUCCESS after 0.000s: hack/verify-generated.sh:14: executing 'git diff --exit-code /Users/mustafa/workspace/go/src/openshift/origin/test/extended/util/annotate/generated/' expecting success
SUCCESS after 0.000s: hack/verify-generated.sh:15: executing 'git diff --exit-code /Users/mustafa/workspace/go/src/openshift/origin/test/extended/util/image/zz_generated.txt' expecting success
[INFO] hack/verify-generated.sh exited with code 0 after 00h 00m 47s
```

**Comment by Elbehery** [comment]
@hasbro17  PTAL needs ` Not mergeable. Needs approved, backport-risk-assessed labels`

**Comment by hasbro17** [comment]
@dgoodwin If you could please add the cherry-pick-approved label 🙇

**Pull request merged**